projects/optic/src/commands/diff/diff-all.ts

```diff
@@ -0,0 +1,365 @@
+        };
+
+        const compressedData = compressData(
+          fromParseResults,
+          toParseResults,
+          specResults,
+          meta
+        );
+        const anonymousId = await getAnonId();
+        trackEvent('optic.diff_all.view_web', anonymousId, {
+          compressedDataLength: compressedData.length,
+        });
+        await open(`${config.client.getWebBase()}/cli/diff#${compressedData}`, {
+          wait: false,
+        });
+      }
+    } else if (
+      !toParseResults.isEmptySpec &&
+      typeof toParseResults.jsonLike[OPTIC_URL_KEY] !== 'string'
+    ) {
+      warnings.missingOpticUrl.push({
+        path: candidate.to!,
+      });
+      continue;
+    }
+  }
+  return {
+    warnings,
+    results,
+  };
+}
+
+type Results = (Awaited<ReturnType<typeof compute>> & {
+  from?: string;
+  to?: string;
+})[];
+
+type Warnings = {
+  missingOpticUrl: {
+    path: string;
+  }[];
+  unparseableFromSpec: {
+    path: string;
+    error: unknown;
+  }[];
+  unparseableToSpec: { path: string; error: unknown }[];
+};
+
+const getDiffAllAction =
+  (config: OpticCliConfig) => async (options: DiffAllActionOptions) => {
+    if (config.vcs?.type !== VCS.Git) {
+      logger.error(
+        `Error: optic diff-all must be called from a git repository.`
+      );
+      process.exitCode = 1;
+      return;
+    }
+
+    let compareToCandidates;
+    let compareFromCandidates;
+
+    try {
+      compareToCandidates = await findOpenApiSpecsCandidates(options.compareTo);
+    } catch (e) {
+      logger.error(
+        `Error reading files from git history for --compare-to ${options.compareTo}`
+      );
+      logger.error(e);
+      process.exitCode = 1;
+      return;
+    }
+
+    try {
+      compareFromCandidates = await findOpenApiSpecsCandidates(
+        options.compareFrom
+      );
+    } catch (e) {
+      logger.error(
+        `Error reading files from git history for --compare-from ${options.compareFrom}`
+      );
+      logger.error(e);
+      process.exitCode = 1;
+      return;
+    }
+
+    const candidatesMap = matchCandidates(
+      {
+        ref: options.compareFrom,
+        paths: compareFromCandidates,
+      },
+      {
+        ref: options.compareTo,
+        paths: compareToCandidates,
+      }
+    );
+
+    const { warnings, results } = await computeAll(
+      candidatesMap,
+      config,
+      options
+    );
+
+    if (warnings.missingOpticUrl.length > 0) {
+      logger.info(
+        chalk.yellow(
+          'Warning - the following OpenAPI specs were detected but did not have x-optic-url keys. `optic diff-all` only runs on specs that include `x-optic-url` keys.'
+        )
+      );
+      logger.info(
+        'Run the `optic api add` command to add these specs to optic'
+      );
+      logger.info(warnings.missingOpticUrl.map((f) => f.path).join('\n'));
+      logger.info('');
+    }
+
+    if (warnings.unparseableFromSpec.length > 0) {
```

Comment:
I'm not sure if this should fail with `exitCode=1` - i'm leaning towards no, because if you have an unparsable from spec, you shouldn't fail since there's nothing you can do here, but we cannot run the spec